src/rdkafka.c

```diff
@@ -1103,22 +1103,23 @@ rd_kafka_t *rd_kafka_new (rd_kafka_type_t type, rd_kafka_conf_t *conf,
 
 	call_once(&rd_kafka_global_init_once, rd_kafka_global_init);
 
-
-	if (!conf)
-		conf = rd_kafka_conf_new();
+        /* Use a copy of conf if supplied, otherwise generate a new one so we can always
+         * delete it upon return without additional conditionals
+         */
+        rd_kafka_conf_t* use_conf = (conf ? rd_kafka_conf_dup(conf) : rd_kafka_conf_new());
```

Comment:
nitpick: pointer on variable side, not type side: `rd_kafka_conf_t *use_conf = ..`

Compare:
`my_type_t* a, b;` <-- looks like a and b have same type attr
to
`my_type_t *a, b;` <-- more evident that they do not